src/transaction.rs

```diff
@@ -123,29 +130,31 @@ impl Transaction {
     /// Create a signed transaction from the given `Instruction`.
     fn new_from_instruction(
         from_keypair: &Keypair,
+        contract: Pubkey,
         instruction: Instruction,
         last_id: Hash,
         fee: i64,
     ) -> Self {
-        Self::new_with_userdata_and_instruction(from_keypair, instruction, last_id, fee, vec![])
+        let userdata = serialize(&instruction).expect("serealize instruction");
```

Comment:
speling: "serealize"

Reply:
And all calls to `serialize` can use `unwrap()`, just like ReadWrite locks. An error indicates a programatic error.